src/components/Arpscan/Arpscan.tsx

```diff
@@ -0,0 +1,208 @@
+import { useState, useCallback, useEffect } from "react";
+import { Button, Stack, TextInput } from "@mantine/core";
+import { useForm } from "@mantine/form";
+import { CommandHelper } from "../../utils/CommandHelper";
+import ConsoleWrapper from "../ConsoleWrapper/ConsoleWrapper";
+import { SaveOutputToTextFile_v2 } from "../SaveOutputToFile/SaveOutputToTextFile";
+import { LoadingOverlayAndCancelButton } from "../OverlayAndCancelButton/OverlayAndCancelButton";
+import { checkAllCommandsAvailability } from "../../utils/CommandAvailability";
+import InstallationModal from "../InstallationModal/InstallationModal";
+import { RenderComponent } from "../UserGuide/UserGuide";
+
+/**
+ * Represents the form values for the ArpScan component.
+ */
+interface FormValuesType {
+    interface: string;
+}
+
+/**
+ * The ArpScan component.
+ * @returns The ArpScan component.
+ */
+function ArpScan() {
+    // Component State Variables.
+    const [loading, setLoading] = useState(false); // State variable to indicate loading state.
+    const [output, setOutput] = useState(""); // State variable to store the output of the command execution.
+    const [pid, setPid] = useState(""); // State variable to store the process ID of the command execution.
+    const [allowSave, setAllowSave] = useState(false); // State variable to allow saving the output to a file.
+    const [hasSaved, setHasSaved] = useState(false); // State variable to indicate if the output has been saved.
+    const [isCommandAvailable, setIsCommandAvailable] = useState(false); // State variable to check if the command is available.
+    const [opened, setOpened] = useState(!isCommandAvailable); // State variable to check if the installation modal is open.
+    const [loadingModal, setLoadingModal] = useState(true); // State variable to indicate loading state for the installation modal.
+
+    // Component Constants.
+    const title = "ArpScan"; // Title of the component.
+    const description = "ArpScan is a tool for scanning the local network for ARP entries."; // Description of the component.
+    const steps =
+        "Step 1: Enter the network interface to scan (e.g., eth0).\n" +
+        "Step 2: Click the 'Start ArpScan' button to initiate the scanning process.\n" +
+        "Step 3: View the output block below to see the discovered ARP entries on the local network.\n";
+    const sourceLink = ""; // Link to the source code or Kali Tools page.
+    const tutorial = ""; // Link to the official documentation/tutorial.
+    const dependencies = ["arp-scan"]; // Dependencies required for the ArpScan tool.
+
+    // Form hook to handle form input.
+    const form = useForm<FormValuesType>({
```

Comment:
We use useForm<FormValuesType> instead of just useForm() to provide type safety and tooling support when working with the form values object in a TypeScript React application.